groups/views.py

```diff
@@ -126,3 +134,30 @@ def form_valid(self, form):
 
     def get_success_url(self):
         return self.discussion.get_absolute_url()
+
+
+class CommentDelete(DeleteView):
+    """
+    Deletes a particular comment after confirming with an 'Are you sure?' page.
+
+    Rather than fully deleting the comment, set its state to DELETED so it continues to
+    exist but displays differently.
+    """
+    model = models.Comment
+    template_name = 'groups/comment_delete.html'
+
+    def dispatch(self, request, *args, **kwargs):
+        """Disallow access to any user other than admins or the comment creator."""
+        self.comment = self.get_object()
+        if not self.comment.may_be_deleted(request.user):
+            message = 'A user may only delete their own comments.'
```

Comment:
Message not entirely true, as some users may be able to delete other's comments.
Perhaps "You do not have permission to delete this comment" would be more fitting.